lnd_test.go

```diff
@@ -2426,6 +2428,174 @@ func testNodeSignVerify(net *networkHarness, t *harnessTest) {
+	// Alice should send all money form her side to Bob.
+	ctxt, _ := context.WithTimeout(ctxb, timeout)
+	chanPoint := openChannelAndAssert(ctxt, t, net, net.Alice, net.Bob,
+		paymentAmt*2000, 0)
+
+	info, err := getChanInfo(net.Alice)
+	if err != nil {
+		t.Fatalf("unable to get alice channel info: %v", err)
+	}
+
+	// Calculate the number of invoices.
+	numInvoices := int(info.LocalBalance / paymentAmt)
+	bobAmt := int64(numInvoices * paymentAmt)
+	aliceAmt := info.LocalBalance - bobAmt
+
+	// Send one more payment in order to cause insufficient capacity error.
+	numInvoices += 1
+
+	// Initialize seed random in order to generate invoices.
+	rand.Seed(time.Now().UnixNano())
+
+	// With the channel open, we'll create a invoices for Bob that
+	// Alice will pay to in order to advance the state of the channel.
+	bobPaymentHashes := make([][]byte, numInvoices)
+	for i := 0; i < numInvoices; i++ {
+		preimage := make([]byte, 32)
+		rand.Read(preimage)
+		invoice := &lnrpc.Invoice{
+			Memo:      "testing",
+			RPreimage: preimage,
+			Value:     paymentAmt,
+		}
+		resp, err := net.Bob.AddInvoice(ctxb, invoice)
+		if err != nil {
+			t.Fatalf("unable to add invoice: %v", err)
+		}
+
+		bobPaymentHashes[i] = resp.RHash
+	}
+
+	// Wait for Alice to receive the channel edge from the funding manager.
+	ctxt, _ = context.WithTimeout(ctxb, timeout)
+	err = net.Alice.WaitForNetworkChannelOpen(ctxt, chanPoint)
+	if err != nil {
+		t.Fatalf("alice didn't see the alice->bob channel before "+
+			"timeout: %v", err)
+	}
+
+	// Open up a payment stream to Alice that we'll use to send payment to
+	// Bob. We also create a small helper function to send payments to Bob,
+	// consuming the payment hashes we generated above.
+	alicePayStream, err := net.Alice.SendPayment(ctxb)
+	if err != nil {
+		t.Fatalf("unable to create payment stream for alice: %v", err)
+	}
+
+	// Send payments from Alice to Bob using of Bob's payment hashes
+	// generated above.
+	now := time.Now()
+	for i := 0; i < numInvoices; i++ {
+		sendReq := &lnrpc.SendRequest{
+			PaymentHash: bobPaymentHashes[i],
+			Dest:        net.Bob.PubKey[:],
+			Amt:         paymentAmt,
+		}
+
+		if err := alicePayStream.Send(sendReq); err != nil {
+			t.Fatalf("unable to send payment: "+
+				"%v", err)
+		}
+	}
+
+	receivedError := false
+	for i := 0; i < numInvoices; i++ {
+		if _, err := alicePayStream.Recv(); err != nil {
+			if strings.Contains(err.Error(), "Insufficient") {
+				receivedError = true
+				continue
+			}
+			t.Fatalf("unable to send payment: %v", err)
+		}
+	}
+
+	if !receivedError {
+		t.Fatalf("insufficient capacity error haven't been received")
+	}
+
+	// All payments have been sent, mark the finish time.
+	timeTaken := time.Since(now)
+
+	// Next query for Bob's and Alice's channel states, in order to confirm
+	// that all payment have been successful transmitted.
+	time.Sleep(time.Second * 3)
+	bobChan, err := getChanInfo(net.Bob)
```

Comment:
Rather than sleeping for three seconds here, we could instead use the invoice subscription streaming RPC to consume all the required notifications (as bob settles each HTLC).